### ISSUE TITLE
Yjs as the source of truth

### ISSUE THREAD

**Issue by lihebi** [codepod-io/codepod#331]
TL;DR Yjs should write the database.

---

The current architecture is centered around the browser (`yjs <-> browser <-> DB`):

- browser (ui/) talk to yjs for peer-syncing
- browser (ui/) talk to DB for data persistence

Since multiple users can have multiple instance of browsers accessing the same repo, this approach has several problems:
- we have some complicated logic to handle when and which user should write the DB.
- there might be consistency issue between Yjs and DB due to bugs

The ideal workflow should be centered around Yjs (`browser <-> Yjs <-> DB`):

- browser talk to yjs for peer-syncing
- Yjs talk to DB for data persistence
